forwards_paper.tex

```diff
@@ -854,35 +871,45 @@ \subsubsection*{Computational complexity}
 Similarly, not every new edge derives from a never-before-seen node,
 but the number of nodes is at most equal to the number of edges plus the sample size.
 The space complexity after simplification is therefore $O(N + N\log N)$, since
-we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
+we need $O(N)$ space to store the first marginal tree, and a constant amount of space
 for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
-to store the complete history of the simulation, but only $O(N + N \log N)$
+to store the complete history of the simulation, but only $O(N \log N)$ % $O(N + N \log N)$
 space to store the history that is relevant to the extant population.
 
 %% Progressive memory reduction as a function of T
 \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
 by retaining only 8 generations of history.
 This begs the question: how fast does the information in the pedigree become irrelevant?
-Suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
+Suppose instead that we have run a simulation of $N$ individuals for $T$ generations,
 begun with no prior history.
 One way to answer this question is to ask:
-How many of the resulting $4NT$ edges are required after simplification?
-As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
-Roughly, the length of time for which a marginal tree has $k$ tips is
-$2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations,
-and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
-This implies that after running for time $T$ we expect to
-have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+How many of the resulting $2NT$ edges are required after simplification?
+As above, the expected number of edges is bounded by $N-2$
+plus four times the sum of the lengths of all marginal trees,
+because new edges appear with most recombinations that occur on any marginal tree.
+Therefore, we need to compute the mean total length of all branches in a coalescent tree
+up until time $T$ (or the common ancestor, whichever comes first).
+Since $k$ lineages coalesce at rate $\binom{k}{2}/N$,
+the expected length of time for which a marginal tree has $k$ branches is
+$N/(k(k-1)) = N(1/(k-1) - 1/k)$ generations.
+Since there are $k$ branches for this time,
+this period is expected to contribute $N/(k-1)$ generations to the total tree length.
```

Comment:
I don't follow the argument here --- what does total tree length have to do with it? I thought we were talking about time rather than total tree length?

Reply:
It is total tree length that tells us about the number of edges, because edges ~ recombs ~ total tree length. I tried to lay this out in the first paragraph; I should refer back to it better.  First we had to work out how (number of roots) relates to time because it's esaier to work out total tree length up until a certain number of roots than to work out total tree length up to a certain time.

Reply:
I followed the bit about counting roots and computing the total tree length, but what threw me was the switching back and forth between the two. I'd be easier if we first computed the number of roots, and then stated the total tree length I think.

Reply:
gotcha; I can rearrange.